src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -47,66 +54,64 @@ def set_default_thresholds(self, default_threshold_reference: float, default_thr
         ori_shape: Tensor,
         threshold: Tensor | None = None,
         num_bg_points: Tensor | None = None,
-    ) -> tuple[Tensor, Tensor]:
+        device: str | torch.device = "cpu",
+    ) -> tuple[dict[int, Tensor], dict[int, Tensor]]:
         """Get prompt candidates."""
         if threshold is None:
             threshold = torch.tensor([[0.0]], dtype=torch.float32)
         if num_bg_points is None:
             num_bg_points = torch.tensor([[1]], dtype=torch.int64)
 
-        device = image_embedding.device
         threshold = threshold.to(device)
 
-        total_points_scores: Tensor = torch.zeros(max(used_indices) + 1, 0, 3, device=device)
-        total_bg_coords: Tensor = torch.zeros(max(used_indices) + 1, num_bg_points, 2, device=device)
-        for label in used_indices:
-            points_scores, bg_coords = self._get_prompt_candidates(
-                image_embedding=image_embedding,
+        total_points_scores: dict[int, Tensor] = {}
+        total_bg_coords: dict[int, Tensor] = {}
+        for label in map(int, used_indices):
+            points_scores, bg_coords = self(
+                image_embeddings=image_embeddings,
                 reference_feat=reference_feats[label],
                 ori_shape=ori_shape,
                 threshold=threshold,
                 num_bg_points=num_bg_points,
-                device=device,
             )
 
-            pad_size = torch.tensor(points_scores.shape[0] - total_points_scores.shape[1])
-            pad_tot = torch.max(self.zero_tensor, pad_size)
-            pad_cur = torch.max(self.zero_tensor, -pad_size)
-
-            total_points_scores = F.pad(total_points_scores, (0, 0, 0, pad_tot, 0, 0), value=-1)
-            points_scores = F.pad(points_scores, (0, 0, 0, pad_cur), value=-1)
-
             total_points_scores[label] = points_scores
             total_bg_coords[label] = bg_coords
 
         return total_points_scores, total_bg_coords
 
-    def _get_prompt_candidates(
+    def forward(
         self,
-        image_embedding: Tensor,
+        image_embeddings: Tensor,
         reference_feat: Tensor,
         ori_shape: Tensor,
-        threshold: Tensor,
-        num_bg_points: Tensor,
-        device: torch.device | str = "cpu",
+        threshold: Tensor | None = None,
+        num_bg_points: Tensor | None = None,
     ) -> tuple[Tensor, Tensor]:
         """Get prompt candidates from given reference and target features."""
-        assert threshold.dim() == 2 and num_bg_points.dim() == 2  # noqa: PT018, S101
+        if threshold is None:
+            threshold = torch.tensor([[0.0]], dtype=torch.float32)
+        if num_bg_points is None:
+            num_bg_points = torch.tensor([[1]], dtype=torch.int64)
+
+        ori_shape = ori_shape.squeeze()
+        threshold = threshold.squeeze()
+        num_bg_points = num_bg_points.squeeze()
 
-        target_feat = image_embedding.squeeze()
+        target_feat = image_embeddings.squeeze()
```

Comment:
Can you add comments to inform what is the desired tensor shape after squeeze?

Reply:
I removed unnecessary operations for the variables such as `ori_shape`, `threshold`, and `num_bg_points` and added the commend for `image_embeddings`' shape at https://github.com/openvinotoolkit/training_extensions/pull/2955/commits/f1aa67c97de4ce745533c2b3745d2e6770974c81.